tlatools/org.lamport.tlatools/src/tlc2/tool/impl/InvariantViolation.java

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2025, Oracle and/or its affiliates.
+
+package tlc2.tool.impl;
+
+import tla2sany.semantic.SemanticNode;
+import tlc2.util.Context;
+
+/**
+ * Information about an invariant violation.  This class has no particular structure, only a single
+ * {@link #formatPretty()} method to display the information.  Today it shows a location of a relevant FALSE
+ * expression and the values of any {@code \A}-bound variables.
+ */
+public class InvariantViolation {
```

Comment:
The naming is inconsistent, using "InvariantViolation" in one place and "find ... Context" in another. It might be clearer to rename "InvariantViolation" to "Assignment" (after verifying it doesn’t conflict with `tlc2.model.Assignment`). Additionally, it’s worth considering whether `InvariantViolation` and `tlc2.value.impl.CounterExample` should be merged into one. Combining them could simplify the process of including this information when counterexamples are serialized in various formats (such as JSON or TLA+).

Reply:
I agree. I'll see if I can come up with something better.